gpkit/constraints/geometric_program.py

```diff
@@ -189,8 +190,15 @@ def _get_solver(solver):
                   ((time() - tic) / soltime * 100))
             tic = time()
 
-        self.check_solution(self.result["cost"], solver_out['primal'],
-                            nu=solver_out["nu"], la=solver_out["la"])
+        if warn_on_check:
+            try:
+                self.check_solution(self.result["cost"], solver_out['primal'],
+                                    nu=solver_out["nu"], la=solver_out["la"])
+            except RuntimeWarning, e:
+                print "Solution check warning", str(e)
+        else:
+            self.check_solution(self.result["cost"], solver_out['primal'],
+                                nu=solver_out["nu"], la=solver_out["la"])
```

Comment:
seems to me this could be implemented more cleanly as:

```python
try:
    self.check_solution(...)
except RuntimeWarning as e:
    if warn_on_check:
        print "Solution check warning", str(e)
    else:
        raise e
```